sky_alert/endpoint.py

```diff
@@ -7,6 +14,73 @@
 def healthz():
     return 'OK'
 
-@app.get('/sunrise')
-def sunrise():
-    return '06:00'
\ No newline at end of file
+@app.get('/openweather_sun_data')
+def sun_data(lat: int = 0, lon: int = 0):
+
+    headers = {
+        'Content-Type': 'application/json',
+    }
+
+    params = {
+        "lat": lat,
+        "lon": lon,
+        "appid": os.environ["OPENWEATHER_API_KEY"]
+    }
+
+    api_url = os.environ["OPENWEATHER_API_URL"]
+
+    try:
+        res = requests.get(api_url, params=params, headers=headers)
+        
+        if res.status_code == 200:
+            json_data = res.json()
+            sun_data = get_sunrise_sunset_from_json(json_data)
+            return sun_data
+
+        elif res.status_code == 401:
+            print("error: Unauthorized")
+            return 401
+
+        elif res.status_code == 404:
+            print("error: Resource not found")
+            return 404
+
+        else:
+            print("error: Internal Server Error")
+            return 500
+    
+    except requests.RequestException as e:
+        print("Request Error")
+        return 400
+    
+    except KeyError as e:
+        print("Key Error")
+        return 400
+    
+    except Exception as e:
+        print("Internal Server Error")
+        return 500
+
+def get_sunrise_sunset_from_json(json_data):
+
+    sunrise = None
+    sunset = None
+
+    # Check for key existence at different levels
+    if 'current' in json_data and isinstance(json_data['current'], dict):
+        current_data = json_data['current']
+        sunrise = current_data.get('sunrise')
+        sunset = current_data.get('sunset')
+
+    # Perform the necessary operations if sunrise and sunset are available
+    if sunrise is not None and sunset is not None:
+        sunrise_datetime = datetime.datetime.utcfromtimestamp(sunrise)
+        sunset_datetime = datetime.datetime.utcfromtimestamp(sunset)
+        return SunData(sunrise_datetime, sunset_datetime)
+
+    else:
+        
+        # Handle missing keys or structure change
+        # Perform necessary actions like setting default values or logging the issue
+        print("Error: Change in API response - consult documentation")
+        return None
```

Comment:
since the calling method `sun_data()` is catching errors, would it be clearer if we throw a specific error, such as `KeyError` or a general `Exception`? This would make sure that the upstream caller would either get a valid `SunData` class back or an error, rather than needing to do another check...